### PR TITLE
Enable functionality for uma and reality modules

### PR DESCRIPTION
**Pull request by abg4**
For context, we started off leveraging this Snapshot branch (https://github.com/snapshot-labs/snapshot/compare/develop...mkt/safesnap-uma-integration) that adds a `moduleAddress` and `moduleType` to the `plugin.json` file (https://github.com/snapshot-labs/snapshot/blob/mkt/safesnap-uma-integration/src/plugins/safeSnap/plugin.json). The `plugin.json` file is the SafeSnap config found in Settings for Snapshot spaces which configures reality and uma modules. 

My PR changes back a lot of the old naming (`umaAddress` to `realityAddress`) along with using `realityAddress` in the `plugin.json` file and then adding an option to include an `umaAddress`. This is to avoid issues with existing and past reality modules/proposals. Adding a `moduleType` and changing `realityAddress` to `moduleAddress` created problems where the old proposal's module address was being overwritten with the address from the `plugin.json`. It might help to look at these changes from the Snapshot branch but I think this was one of the main reasons this branch was discontinued and this functionality was rolled into the refactoring branch:
- Updating this `coerceConfig` function: https://github.com/snapshot-labs/snapshot/compare/develop...mkt/safesnap-uma-integration#diff-ee2c202cf8068f1267a1521322ce06a52eb6d1679de7755250eae6561ca53d1fR109
- Updates to the `Proposal.vue` file: https://github.com/snapshot-labs/snapshot/compare/develop...mkt/safesnap-uma-integration#diff-722ab97ab00c96e1c5e6404d5be68112e5f0b8f6e134d5f0d8cf00c40f434f5eR12

In this PR I attempted to keep the ‘realityAddress’ and maintain as much of the current Snapshot functionality until the transaction execution where the `umaAddress` is checked if a valid optimistic governor contract is included in the config. If there is a valid optimistic governor, a separate handleOutcome file is used.

TODO: Right now we created a `HandleOutcomeReality.vue` file and have UMA modules use the `HandleOutcome.vue` file. Before submitting a PR to Snapshot, it might make it easier to review and decrease the number of changes if we keep reality using the original `HandleOutcome.vue` file and create a `HandleOutcomeUma.vue` file for UMA modules.

### CODE REVIEW

**Comment by pemulis** [comment]
I agree with your instinct in the TODO. Is there a reason you aren't making that change right now? If it's just to avoid conflicting with the branch I'm working on, it will be an easy change. I'm inclined to get it into this PR if possible so that the last thing we need to do before submitting the PR to Snapshot is the minor clean-up stuff I'm doing related to UMA proposals.

**Review by pemulis — COMMENTED** [review]
There are a few locations (`index.ts`, `interfaces.ts`, `SafeTransactions.vue`, `Proposal.vue`, etc.) where an `uma` address is replaced with a `reality` address. Does that not present any problems in the UMA-specific `HandleOutcome` flow?

**Review by pemulis — APPROVED** [review]

**Comment by abg4** [comment]
@pemulis I agree it's better just to include the `HandleOutcome` file changes in this PR and my latest commit handles the update.

Good question on the naming from `uma` to `reality`. I have been checking the `HandleOutcomeUma` props and haven't seen anything that has been impacted or that would be impacted by the name changing. I want to keep testing different scenarios before submitting a PR to Snapshot but the goal of changing the naming back to reality is to prevent the number of changes and avoid any possible issues for DAO's currently using reality modules.

File: src/plugins/safeSnap/components/HandleOutcome.vue [hunk 1]
```diff
@@ -16,6 +16,8 @@ import {
   useSafe
 } from '@/composables';
 
+import SafeSnapModalOptionApproval from './Modal/OptionApproval.vue';
```
Comment (mrice32) [inline src/plugins/safeSnap/components/HandleOutcome.vue:19]: I didn't carefully review the two *Outcome.vue files, I just assume that this file is the original version (pre UMA changes) and the other file (that's all additions) is the post-uma changes that have already been reviewed. Lmk if it needs more careful review!

File: src/plugins/safeSnap/index.ts [hunk 2]
```diff
@@ -82,13 +92,79 @@ export default class Plugin {
...
+      provider
+    );
+
+    let rules;
+    try {
+      rules = await moduleContract.rules();
+    } catch {
+      rules = undefined;
+    }
+    return rules !== undefined ? 'uma' : 'reality';
```
Comment (mrice32) [inline src/plugins/safeSnap/index.ts:158]:
```suggestion
    return moduleContract.rules().then(() => 'uma').catch(() => 'reality');
```

**Review by mrice32 — APPROVED** [review]
LGTM!

File: src/plugins/safeSnap/components/HandleOutcome.vue (hunk 1 above)
Comment (abg4) [reply src/plugins/safeSnap/components/HandleOutcome.vue:19]: Yep, it was just the naming that changed for these files. I confirmed `HandleOutcome.vue` matches the original version (pre-UMA changes) and `HandleOutcomeUma.vue` includes all our changes. John and I will do another check during our clean-up/review so I don't think any more review on these files is necessary. Thanks for reviewing!

**Pull request merged**